### DISCUSSION TITLE
Sample size for GradientExplainer

### DISCUSSION THREAD

**Discussion by andresalfp** [shap/shap#2293] [Q&A]
Hello,

I'm trying to calculate Shapley values with GradientExplainer, but I have no idea what is the correct sample size for the estimation. I have a dataset with 1100 observations, each with 12 explainer features.

Is there any rule of thumb to select a proper sample size?

Greetings,

Andrés

**Comment by jainraj** [comment]
Check [Improving performance of deep learning models with axiomatic attribution priors and expected gradients - Erion et al.](https://arxiv.org/pdf/1906.10670.pdf) which **introduced the Gradient Explainer** method. Section C.1 says that the authors found ~200 background samples were enough. You could use that. I personally would suggest using any higher number as long as the computation time isn't too large. [As we know, more samples, better estimates]. 

For 12 features, I am thinking all 1100 could be used. Maybe you can try and let people know here about the computation time.
